save-core/build.gradle.kts

```diff
@@ -76,6 +75,17 @@ tasks.withType<KotlinCompile<*>>().forEach {
     it.dependsOn(generateVersionFileTaskProvider)
 }
 
+val taskNames = listOf("sourcesJar", "compileCommonNonJsMainKotlinMetadata")
```

Comment:
Starting from `gradle 8x` these tasks should have explicit dependencies on `generateConfigOptionsTaskProvider` `generateVersionFileTaskProvider`, however, with current realization these tasks is not yet created at this moment, therefore `mustRunAfter` is not applies

Reply:
may be someone else can beat it

Reply:
Apparently, we can feed `Provider` into `SourceDirectorySet#srcDir`, thus indicating dependency between tasks. I've added this in a842672